game-core/src/main/java/games/strategy/engine/data/MoveDescription.java

```diff
@@ -0,0 +1,80 @@
+package games.strategy.engine.data;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Maps;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.Getter;
+
+/** Describes an action that moves one or more units along a specific route. */
+public class MoveDescription extends AbstractMoveDescription {
+  private static final long serialVersionUID = 2199608152808948043L;
+  @Getter private final Route route;
+  @Getter private final Map<Unit, Unit> unitsToTransports;
+  @Getter private final Map<Unit, Collection<Unit>> dependentUnits;
+
+  public MoveDescription(
+      final Collection<Unit> units,
+      final Route route,
+      final Map<Unit, Unit> unitsToTransports,
+      final Map<Unit, Collection<Unit>> dependentUnits) {
+    super(Collections.unmodifiableCollection(units));
+    this.route = Preconditions.checkNotNull(route);
+    this.unitsToTransports = Collections.unmodifiableMap(unitsToTransports);
+    this.dependentUnits =
+        Collections.unmodifiableMap(
+            dependentUnits.entrySet().stream()
+                .collect(
+                    Collectors.toMap(
+                        Map.Entry::getKey,
+                        e -> Collections.unmodifiableSet(new HashSet<>(e.getValue())))));
```

Comment:
Just learned that https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html#copyOf(java.util.Collection)

Is a thing, which performs an immutable copy of a set.

Reply:
Thanks, done!